packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,648 @@
+		if (this.showTooltip) {
+			this._tooltipVisibility = "visible";
+		}
+	}
+
+	/**
+	 * Hides the tooltip(s) if the <code>showTooltip</code> property is set to true
+	 *
+	 * @private
+	 */
+	_onmouseout(event) {
+		if (this.showTooltip) {
+			this._tooltipVisibility = "hidden";
+		}
+	}
+
+	/**
+	 * Handle the responsiveness of the Slider's UI elements when resing
+	 *
+	 * @private
+	 */
+	_handleResize() {
+		if (!this.showTickmarks) {
+			return;
+		}
+
+		// Mark resizing to avoid unneccessary calls to that function after rendering
+		this.notResized = false;
+
+		// Convert the string represented calculation expression to a normal one
+		// Check the distance  in pixels exist between every tickmark
+		const spaceBetweenTickmarks = this._spaceBetweenTickmarks();
+
+		// If the pixels between the tickmarks are less than 8 only the first and the last one should be visible
+		// In such case the labels must correspond to the tickmarks, only the first and the last one should exist.
+		if (spaceBetweenTickmarks < 8) {
+			this._tickmarksBackground = `linear-gradient(to right, currentColor 1px, transparent 0) 0 center / calc(100% - 1px) 100% repeat-x`;
+			this._hiddenTickmarks = true;
+			this._labelsOverlapping = true;
+		} else {
+			this._drawDefaultTickmarks(this.step, this.max, this.min);
+			this._hiddenTickmarks = false;
+		}
+
+		if (this.labelInterval <= 0 || this._hiddenTickmarks) {
+			return;
+		}
+
+		// Cache the labels if not yet fetched
+		if (!this._labels) {
+			this._labels = this.shadowRoot.querySelectorAll(".ui5-slider-labels li");
+		}
+
+		// Check if there are any overlapping labels.
+		// If so - only the first and the last one should be visible
+		this._labelsOverlapping = Array.prototype.some.call(this._labels, label => label.scrollWidth > label.clientWidth);
+	}
+
+	/**
+	 * Called when the user starts interacting with the slider.
+	 * After a down event on the slider root, listen for move events on window, so the slider value
+	 * is updated even if the user drags the pointer outside the slider root.
+	 *
+	 * @protected
+	 */
+	handleDownBase(event, min, max) {
+		// Only allow one type of move event to be listened to (the first one registered after the down event)
+		this._moveEventType = !this._moveEventType ? SliderBase.MOVE_EVENT_MAP[event.type] : this._moveEventType;
+
+		SliderBase.UP_EVENTS.forEach(upEventType => window.addEventListener(upEventType, this._upHandler));
+		window.addEventListener(this._moveEventType, this._moveHandler);
+
+		this._boundingClientRect = this.getBoundingClientRect();
+		const newValue = SliderBase.getValueFromInteraction(event, this.step, min, max, this._boundingClientRect, this.directionStart);
+
+		return newValue;
+	}
+
+	/**
+	 * Called when the user finish interacting with the slider
+	 * Fires an <code>change</code> event indicating a final value change, after user interaction is finished.
+	 *
+	 * @protected
+	 */
+	handleUpBase() {
+		this.fireEvent("change");
```

Comment:
Shouldn't we check if there was really a change before firing the change event, for example if the user moves the handle and release it at the same place